linker/resolve.go

```diff
@@ -173,6 +173,11 @@ func (r *result) resolveReferences(handler *reporter.Handler, s *Symbols) error
 		}
 	}
 
+	// This is to de-dupe extendee-releated error messages when the same
+	// extendee is referenced from multiple extension field definitions.
+	// We leave it nil if there's no AST.
+	var extendeeNodes map[ast.Node]struct{}
```

Comment:
Some of the duplicate errors were reported were that any `extends` declaration would report the same error messages about the extended message for every field therein.

So if a block declared three extension fields, it would emit three redundant errors about the extendee. This allows the code to skip reporting the extendee issues for subsequent fields if the extendee maps to a different location in the source code. That way, if the same extendee has the same errors, but in two _different_ `extend` blocks, we can still report all of them without duplicates for a single block.